source/Octopus.Tentacle/Kubernetes/KubernetesEventMonitorTask.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Octopus.Diagnostics;
+using Octopus.Tentacle.Background;
+using Polly;
+
+namespace Octopus.Tentacle.Kubernetes
+{
+    public class KubernetesEventMonitorTask : BackgroundTask
+    {
+        public delegate KubernetesEventMonitorTask Factory(IKubernetesEventMonitor eventMonitor);
+        
+        readonly IKubernetesEventMonitor eventMonitor;
+        readonly ISystemLog log;
+        readonly TimeSpan taskInterval = TimeSpan.FromMinutes(5);
+        public KubernetesEventMonitorTask(ISystemLog log, IKubernetesEventMonitor eventMonitor) : base(log, TimeSpan.FromSeconds(30))
+        {
+            this.log = log;
+            this.eventMonitor = eventMonitor;
+        }
+        
+        protected override async Task RunTask(CancellationToken cancellationToken)
+        {
+            if (!KubernetesConfig.MetricsIsEnabled)
+            {
+                log.Info("Event monitoring for agent metrics is not enabled.");
+                return;
+            }
+            
+            //We don't want the monitoring to ever stop
+            var policy = Policy.Handle<Exception>().WaitAndRetryForeverAsync(
+                retry => TimeSpan.FromMinutes(5),
```

Comment:
Maybe use the `taskInterval` value here. It makes sense because it doesn't matter if it's successful or failure, we always want to wait <taskInterval> time before trying again.